src/ResourceManager/Compute/Commands.Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -174,6 +183,21 @@ public partial class NewAzureRmVmss : ComputeAutomationBaseCmdlet
 
             imageAndOsType = await client.UpdateImageAndOsTypeAsync(ImageName, Location);
 
+            NatBackendPort = imageAndOsType.OsType.UpdatePorts(NatBackendPort);
+
+            var inboundNutRuleName = VMScaleSetName;
+            var portRangeStart = 50000;
+            foreach (var natBackendPort in NatBackendPort)
+            {                
+                var inboundNutRule = loadBalancer.CreateInboundNatPool(
+                    name: inboundNutRuleName + natBackendPort.ToString(),
+                    frontendIpConfiguration: frontendIpConfiguration,
+                    frontendPortRangeStart: portRangeStart,
+                    frontendPortRangeEnd: portRangeStart + InstanceCount,
+                    backendPort: natBackendPort);
+                portRangeStart += 1000;
```

Comment:
How many rules could we have?  I wonder if we should just fail gracefully if #instances > 1000

Reply:
#5657